Add numerical support

sfghuis opened the pull request:
!Do not approve!

Sorry for the late response, I am changing jobs and had different priorities. I added support for pydantic, reduced multi-erlangC and erlangC to single ErlangC. Currently implemented test for the single scenario option and will expand on the mutli-scenario one later.

Aim of the mutli-scenario one is to allow for forecast uncertainties to be included as well, granting better insight into potential staffing requirements.

Could really use some feedback on how to improve the setup.

Commit:
remove premature rounding

Commit:
rework for float handeling

sfghuis committed:
pydantic base and multi scenari

sfghuis committed:
expected constants

sfghuis committed:
add pydantic dependency

sfghuis committed:
add pydantic to req

sfghuis committed:
Remove multi erlang

sfghuis committed:
add improved legacy behaviour

sfghuis committed:
basic testing

rodrigo-arenas committed:
Merge pull request #42 from rodrigo-arenas/0.5.1dev

Bump project dependencies

rodrigo-arenas committed:
Merge pull request #43 from rodrigo-arenas/0.5.1dev

release version 0.5.1

sfghuis committed:
fix dependencies

rodrigo-arenas commented:
@sfghuis so far it looks good, I'd just suggest to change the cErlangC_generic_variable_list to a dictionary, this way at the future, we don't rely or relative positions for each variable, but instead we use their names

Let me know if you want me to check something else

sfghuis committed:
finalize basic testing

sfghuis committed:
Merge pull request #1 from sfghuis/add_multi_scenario_test

finalize basic testing

sfghuis commented:
@rodrigo-arenas I will work on making the list a dictionary. I hope to finalize the request today

sfghuis committed:
create dict

sfghuis committed:
Merge branch 'main' of https://github.com/sfghuis/pyworkforce into rework-var-list-to-dict

sfghuis committed:
rename erlangcdata class

sfghuis committed:
Merge pull request #2 from sfghuis/rework-var-list-to-dict

create dict

sfghuis committed:
Merge branch 'main' repo into fork

sfghuis committed:
update release notes

sfghuis commented:
@rodrigo-arenas Looking forward on your feedback. I added changes notes and updated the dependencies to be inline with your latest version.

Would it be ok if I added some error handeling for overflows values after finishing this request? In cases the transactions and intesities cause overflow errors?

Dsr-Pro commented:
Can we add ErlangX also in a release?
